oscarlab/graphene
# [Pal/LibOS] Add per-directory vimrc files

Please fill in the following form before submitting this PR and ensure that your code follows our [coding style guideline](../blob/master/CODESTYLE.md).

## Affected components


## Description of the changes (reasons and measures)

This PR adds per-directory vimrc files to major source directories. Unlike emacs, vim only loads .vimrc from the user's home directory and the current working directory. I cannot find other ways that do not require our contributors to install extra plugins.

Note: the scripts won't work if users open from a directory that has no .vimrc file.

After merging this PR, #324 is ready to merge.

## How to test this PR? (if applicable)

Open a source file with vim and type `:set textwidth?` If the local .vimrc file is loaded, you should see `textwidth=100`.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/oscarlab/graphene/530)
